### PR TITLE
CORGI 507 - Update concourse resource to work with update CORGI

### PR DESCRIPTION
**Pull request by TylerZeroMaster**
- [ ] openstax/corgi/issues/507

Modify corgi resource to translate between the old and new data models.

### CODE REVIEW

**Review by ckline-tryptic — APPROVED** [review]

File: corgi-concourse-resource/corgi_concourse_resource/corgi_api.py [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 
-def build_url(api_root, *args):
+def build_url(api_root, *args, **kwargs):
     parts = [api_root]
+    query = []
     parts.extend(args)
     parts = [str(p) for p in parts]
-    return "/".join(parts)
+    url = "/".join(parts)
+    for item in kwargs.items():
```
Comment (philschatz) [inline corgi-concourse-resource/corgi_concourse_resource/corgi_api.py:14]: Why not [urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode)?

File: corgi-concourse-resource/corgi_concourse_resource/utils.py [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
 
 
+def get_collection_id(job):
+    repo = job["repository"]
+    book_slug = job["books"][0]["slug"]
+    owner = repo["owner"]
+    repo_name = repo["name"]
+    if owner != "openstax":  # pragma: no cover
+        return f"{owner}/{repo_name}/{book_slug}"
+    return f"{repo_name}/{book_slug}"
```
Comment (philschatz) [inline corgi-concourse-resource/corgi_concourse_resource/utils.py:12]:
```suggestion
    return f"{owner}/{repo_name}/{book_slug}"
```

Why special-case openstax? This way the "collection_id" must always have _exactly_ 2 slashes in it :joy:

File: corgi-concourse-resource/corgi_concourse_resource/utils.py [hunk 3]
```diff
@@ -2,6 +2,20 @@
...
+
+
+def get_style(job):
+    return job["books"][0]["style"]
```
Comment (philschatz) [inline corgi-concourse-resource/corgi_concourse_resource/utils.py:16]: (not important, just for discussion) With the new CORGI isn't this always `default`? I thought we decided that if folks want to test with different recipes they can create a branch, no?

File: build-concourse/README.md [hunk 4]
```diff
@@ -128,25 +128,97 @@ Upload the following (tweak it to change which jobs are pulled) by visiting http
...
+        "repository": {
+          "name": "osbooks-college-success",
+          "owner": "openstax"
+        },
+        "job_type": {
+          "name": "git-web-hosting-preview",
+          "display_name": "Web Preview (git)",
+          "id": "4"
+        },
+        "user": {
```
Comment (philschatz) [inline build-concourse/README.md:152]: Wow, that's a lot of extra JSON... is that all used by the resource or is this API endpoint now shared between concourse and the UI?

If it's shared, are all of these fields necessary for concourse?

**Review by philschatz — COMMENTED** [review]
Just a few questions but looks good in general

File: corgi-concourse-resource/corgi_concourse_resource/utils.py (hunk 2 above)
Comment (TylerZeroMaster) [reply corgi-concourse-resource/corgi_concourse_resource/utils.py:12]: I will give that a try. Some of this was just me mindlessly copying and making necessary edits.

File: corgi-concourse-resource/corgi_concourse_resource/utils.py (hunk 3 above)
Comment (TylerZeroMaster) [reply corgi-concourse-resource/corgi_concourse_resource/utils.py:16]: Yep, this is only used to pass the style to the Enki pipeline. I guess the pipeline parses it from the books.xml too, so we can probably do away with this property in the input.

You can see an example of what is fetched at `corgi-pr-514-enki/jobs/build-pdf/builds/169` in concourse.

File: build-concourse/README.md (hunk 4 above)
Comment (TylerZeroMaster) [reply build-concourse/README.md:152]: It has always been shared. In the past, the saved responses in the README omitted much of the included information. When I updated the README, I thought I wanted to use data that was true-to-life.

File: corgi-concourse-resource/corgi_concourse_resource/corgi_api.py (hunk 1 above)
Comment (TylerZeroMaster) [reply corgi-concourse-resource/corgi_concourse_resource/corgi_api.py]: I overlooked that. Thanks, I am using it instead now.

**Review by philschatz — APPROVED** [review]

**Pull request merged**